pinc/SiteConfig.inc

```diff
@@ -0,0 +1,144 @@
+<?php
+
+class SiteConfig
+{
+    private static bool $_is_loaded = false;
+
+    // alert messages
+    public static bool $maintenance = false;
```

Comment:
I wonder if we could make the `SiteConfig` object immutable.

Unfortunately, I don't think this is possible with all the fields `static` as we would need a specific instance. This could be done through a singleton (e.g. `SiteConfig::get()`) and making the fields `readonly`, this would be a slightly longer than the current scheme but it would enforce that the fields can't be changed mid-request.

Reply:
I originally started with an object-based scheme which meant accessing things was `SiteConfig::get()->value` which seemed more unwieldy than `SiteConfig::$value` when I started using it.

re: immutability - we'd need to think about how to make default values work since we can't set them and override them. I can probably figure that out. We'll also need to figure out how to handle some unit test that do need to override these (which is decidedly an anti-pattern but also where we are right now).

It's a good suggestion, let me noodle over it.

Reply:
I think I've figured out how to make default values work with `readonly`, but as I think more about our unit tests and wanting to test variations I'm unclear how we do that without either making the config modifiable at runtime or allowing the creation, and somehow use, of a new SiteConfig object.

I'm not sure enforcing immutability is important enough for the effort. Thoughts?